src/ProjectTemplates/BlazorWasm.ProjectTemplates/content/BlazorWasm-CSharp/Client/Services/IdentityAuthenticationStateProvider.cs

```diff
@@ -0,0 +1,59 @@
+using BlazorWasm_CSharp.Shared.Models;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Authorization;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace BlazorWasm_CSharp.Client
+{
+    public class IdentityAuthenticationStateProvider : AuthenticationStateProvider
+    {
+        private ClaimsPrincipal _currentUser;
+
+        public IdentityAuthenticationStateProvider(HttpClient client)
+        {
+            Client = client;
+        }
+
+        public HttpClient Client { get; }
+
+        public override async Task<AuthenticationState> GetAuthenticationStateAsync()
+        {
+            if (_currentUser != null)
+            {
+                return new AuthenticationState(_currentUser);
+            }
+
+            await UpdateUser();
+
+            return new AuthenticationState(_currentUser);
+        }
+
+        private async Task UpdateUser()
+        {
+            var user = await Client.GetJsonAsync<UserInfo>("/User");
```

Comment:
```suggestion
            var user = await Client.GetJsonAsync<UserInfo>("User");
```

... because we don't know whether it's mounted at the root of the domain or not, and the `<base href>` should control this.